firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/chunks.txt

```diff
@@ -0,0 +1,103 @@
+    WHEN("the input data exceeds bounds after output is called, without delimiter")
+        THEN("the input status is equal to ok")
+      THEN("the output status is equal to waiting")
+      THEN("the output buffer is empty")
+      THEN("the final input status is equal to invalid_length")
+
+Scenario: ChunkSplitter behaves correctly across chunk boundary(256 bytes)
+
+  GIVEN("A uint8_t chunk splitter of internal buffer equal to 256 bytes")
+    WHEN("enough data is input to exactly fill the buffer of the splitter")
+        THEN("the input status is equal to ok")
+
+  GIVEN("A chunk splitter of buffer size equal to 256 bytes, that is filled with input data")
+    WHEN("the input data exceeds bounds without output being called")
+        THEN("the input status is equal to ok")
+        THEN("the final input status is equal to invalid_length")
+
+  GIVEN("A chunk splitter of buffer size equal to 256 bytes, that is filled with input data")
+    WHEN("the input data exceeds bounds after output is called, without delimiter")
+        THEN("the input status is equal to ok")
+      THEN("the output status is equal to waiting")
+      THEN("the final input status is equal to invalid_length")
+
+  GIVEN("A chunk splitter of buffer size equal to 256 bytes, that is filled with input data")
+    WHEN("Another byte is given as input")
+      THEN("the final input status should be invalid_length")
+
+    WHEN("output is called after the input status returns invalid length")
+      THEN("the output status is equal to invalid_length")
+
+    WHEN("the input receives delimeter")
+      THEN("input status is equal to output_ready")
+      THEN("the output status is equal to ok")
+      THEN("the final input status is equal to ok")
+
+  GIVEN("A chunk splitter of buffer size equal to 256 bytes, that is filled completely with input data")
+    WHEN("delimeter is passed after the buffer is full")
+        THEN("the initial status is equal to ok")
+      THEN("the final input status is equal to output ready")
+
+  GIVEN("A uint8_t chunk splitter with internal buffer of 256 bytes, with delimeter equal to 0x01")
+    WHEN("output is called after input status is ok")
+      THEN("the output status is equal to waiting")
+
+    WHEN("input receives delimeter equal to 1 byte")
+      THEN("the output status is equal to ok")
+      THEN("the input status shall be output ready")
+
+  GIVEN("A uint8_t chunk splitter with internal buffer of 256 bytes, with delimeter equal to 255 bytes")
+    WHEN("output is called after input status is ok")
+      THEN("the output status is equal to waiting")
+    WHEN("input receives delimeter equal to 1 byte")
+      THEN("the input status shall be output ready")
+      THEN("the output status is equal to ok")
+
+  GIVEN("A uint8_t chunk splitter with internal buffer of 256 bytes, with include_delimeter bool equal to true")
+    WHEN("output is called after input status is ok")
```

Comment:
I think the sequence of events in WHEN will be clearer if we write it left to right (this applies to all test definitions in this project):

WHEN one byte is passed as input and output is called
THEN the input method returns ok
THEN the output method returns waiting and sets `output_buffer` to XXXXX